Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1431,11 +1431,12 @@ namespace Babylon
     {
         const auto textureDestination = info[0].As<Napi::Pointer<Graphics::Texture>>().Get();
         const auto textureSource = info[1].As<Napi::Pointer<Graphics::Texture>>().Get();
-
+        
+        // Append a task instead of calling blit immediately or blit happens before drawcalls.
         arcana::make_task(m_update.Scheduler(), *m_cancellationSource, [this, textureDestination, textureSource, cancellationSource = m_cancellationSource]() {
```

Comment:
Did you try removing this? I think we might need to add this to the command queue for the ordering to be correct.

Reply:
Yes, I tried and copy doesn't happen in the correct order when removed.